mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,17 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.deleteOrderById = (req, res) => {
+  Order.deleteOne({ _id: req.params.orderId })
+    .exec()
+    .then(result => {
+      if (result.deletedCount === 0) {
```

Comment:
The condition `if (result.deletedCount === 0)` should be checked for null or undefined before comparing it to 0, as `result.deletedCount` might be null or undefined if the delete operation fails. It should be `if (result && result.deletedCount === 0)`.